File: Dart/src/global.dart
class Global {
  Global._();

  static final Global _instance = Global._();

  factory Global() {
    return _instance;
  }

  void action1() {
    print('\n\r...Action1...');
  }

  void action2() {
    print('\n\r...Action2...');
  }
}

File: Java/src/Global.java
public final class Global {

    // Private constructor prevents external instantiation
    private Global() {}

    // Singleton holder
    private static class Holder {
        private static final Global INSTANCE = new Global();
    }

    // Global access point
    public static Global shared() {
        return Holder.INSTANCE;
    }

    public void action1() {
        System.out.printf("\n\r...Action1...");
    }

    public void action2() {
        System.out.printf("\n\r...Action2...");
    }
}
